tests/scripts/single_card/run_demos_single_card_n300_tests.sh

```diff
@@ -9,8 +32,12 @@ fi
 
 export WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml
 
+# Run tests working on both N150 and N300
 source tests/scripts/single_card/run_demos_single_card_n150_tests.sh
 
+# Falcon7B N300 demo tests
+run_n300_falcon7b_tests
```

Comment:
this is only on n300?

Reply:
Yes, the perf and output tokens may differ for n150 so those would need to be added separately.

Reply:
Sounds good

I would recommend putting some defensive programming in the test to ensure that you're only measuring perf on an X2, such as by checking the number of PCIe devices for example.

@tt-aho @aliuTT We don't have a fixture yet that checks for X1/X2 card types right? Such a fixture would be really useful for everyone

Reply:
Is there an open issue for this? I suppose we can leverage the board types that gets flashed to WH cards
<img width="178" alt="image" src="https://github.com/user-attachments/assets/afa34b77-9a23-44b4-b651-6ca556fc5688">

Reply:
Hmm, do we not expose the cluster descriptor as an API? And especially up to Python level?